substrate/client/consensus/beefy/src/worker.rs

```diff
@@ -348,9 +350,24 @@ impl<B: Block> PersistedState<B> {
 			}
 		}
 
-		if log_enabled!(target: LOG_TARGET, log::Level::Debug) {
-			// verify the new validator set - only do it if we're also logging the warning
-			if verify_validator_set::<B>(&new_session_start, &validator_set, key_store).is_err() {
+		// verify we're not using BEEFY dummy key when role is authority.
+		if is_authority {
+			let mut beefy_key_present = false;
+			for key in key_store.public_keys().unwrap_or_default() {
+				let id_raw: &[u8] = key.as_slice();
+				if id_raw[0..4] != *b"beef" {
+					// Non-dummy key found.
+					beefy_key_present = true;
+				}
+			}
+			if !beefy_key_present {
```

Comment:
```suggestion
			if key_store.public_keys().map_or(false, |k| k.is_empty()) {
```

Reply:
No one is adding these dummy keys to its keystore and even if they have done this, they have done this on purpose manually. 

So, we can just check if they have any key at all for BEEFY.